libraries/api-documenter/src/SimpleDom.ts

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+// ----------------------------------------------------------------------------
+
+/**
+ * A block of plain text, possibly with simple formatting.
+ */
+export interface IDomText {
+  kind: 'text';
+  content: string;
+  bold?: boolean;
+  italics?: boolean;
+}
+
+export type DomCodeHighlighter = 'javascript' | 'plain';
+
+/**
+ * Source code shown in a fixed-width font, with syntax highlighting.
+ */
+export interface IDomCode {
+  kind: 'code';
+  code: string;
+  highlighter: DomCodeHighlighter;
+}
+
+export type DomLinkText = IDomText | IDomCode;
+
+// ----------------------------------------------------------------------------
+
+/**
+ * A block of plain text, possibly with simple formatting.
+ */
+export interface IDomDocLink {
+  kind: 'doc-link';
+  elements: DomLinkText[];
+  targetDocId: string;
+}
+
+/**
+ * A hyperlink to a web page.
+ */
+export interface IDomWebLink {
+  kind: 'web-link';
+  elements: DomLinkText[];
+  targetUrl: string;
+}
+
+/**
+ * A paragraph separator, similar to the "<p>" tag in HTML.
+ */
+export interface IDomParagraph {
+  kind: 'paragraph';
+}
+
+/**
+ * A line break, similar to the "<br>" tag in HTML.
+ */
+export interface IDomLineBreak {
+  kind: 'break';
+}
+
+export type DomBasicText = DomLinkText | IDomDocLink | IDomWebLink | IDomParagraph | IDomLineBreak;
+
+// ----------------------------------------------------------------------------
+
+/**
+ * A top-level heading
+ */
+export interface IDomHeading1 {
+  kind: 'heading1';
+  text: string;
+}
+
+/**
+ * A sub heading
+ */
+export interface IDomHeading2 {
+  kind: 'heading2';
+  text: string;
+}
+
+/**
+ * A box containing source code with syntax highlighting.
+ */
+export interface IDomCodeBox {
+  kind: 'code-box';
+  code: string;
+  highlighter: DomCodeHighlighter;
+}
+
```

Comment:
difference between this and IDomCode ?

Reply:
One of them is a fancy callout box.  The other is `maybe just a word or two` in a code font.

---
In reply to: [135167235](https://github.com/Microsoft/web-build-tools/pull/322#discussion_r135167235) [](ancestors = 135167235)